src/Lean/Compiler/InlineAttrs.lean

```diff
@@ -32,6 +32,26 @@ private def isValidMacroInline (declName : Name) : CoreM Bool := do
     return false
   return true
 
+/--
+Changes the inlining behavior. This attribute comes in several variants:
+- `@[inline]`: marks the definition to be inlined when it is appropriate.
+- `@[inline_if_reduce]`: marks the definition to be inlined if the term after inlining and
+  applying reduction isn't a `match` expression. This attribute can be used for inlining
+  structurally recursive functions. Note: the current/old compiler (written in C++) does not
+  support this attribute.
+- `@[noinline]`: marks the definition to never be inlined.
+- `@[always_inline]`: marks the definition to always be inlined.
+- `@[macro_inline]`: marks the definition to always be inlined early. This makes it possible
```

Comment:
```suggestion
- `@[macro_inline]`: marks the definition to be inlined during elaboration. Such definitions are
  expanded before sending terms to the kernel, rather than in the compiler. This makes it possible
```